packages/retail-ui-extensions/src/extension-api/CartApi/CartApiContent.ts

```diff
@@ -0,0 +1,102 @@
+import type {Address, Cart, Customer, CustomSale} from '../Models/Cart';
+
+export type DiscountType = 'Percentage' | 'FixedAmount' | 'Code';
+
+export interface CartApiContent {
+  /** Initial value of the Cart when the screen is rendered. */
+  initialValue: Cart;
+
+  /** Callback when the cart changes */
+  setOnChange(onChange: (cart: Cart) => void): void;
```

Comment:
Could you please consider using `@remote-ui/async-subscription` (https://github.com/Shopify/remote-ui/tree/main/packages/async-subscription) for any changeable value you are sharing? I designed this based on what I knew we would need for checkout, where everything is mutable, and it handles edge cases you likely are not (handling subscriptions in a way that doesn't ever lose updates due to async timing issues, gracefully managing the retain and release of subscriptions, etc)